### PR TITLE
Delete old V2 offline files

### PR DESCRIPTION
**Pull request by tevincent**

### CODE REVIEW

File: app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt [hunk 1]
```diff
@@ -186,6 +183,19 @@ class MainActivity : BaseActivity() {
...
+
+    // After the migration from API V2 to API V3, offline files were saved in a different folder called "Private". Because we
+    // cannot know if we're in a migration or not, we just delete old files and we marked previously isOffline files as
+    // isMarkedAsOffline to let the BulkDownloadWorker redownload files.
+    private fun deleteOldOfflineFolder() {
+        val userDrive = UserDrive()
+        val offlineFolder =  IOFile(File.getOfflineFolder(this), "${userDrive.userId}/${userDrive.driveId}")
+        offlineFolder.listFiles()?.forEach { file ->
+            if (file.name != "Private") file.deleteRecursively()
+        }
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt:198]: Move to `SyncOfflineUtils`

File: app/src/main/java/com/infomaniak/drive/utils/Utils.kt [hunk 2]
```diff
@@ -366,30 +366,34 @@ object Utils {
...
-                .build()
-
-            workManager.enqueueUniqueWork(BulkDownloadWorker.TAG, ExistingWorkPolicy.APPEND_OR_REPLACE, downloadRequest)
+            enqueueBulkDownloadWorker(context, folderId, userDrive)
 
             onSuccess.invoke()
             emit(FileResult(isSuccess = true))
         }
 
+    fun enqueueBulkDownloadWorker(context: Context, folderId: Int, userDrive: UserDrive = UserDrive()) {
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/utils/Utils.kt:375]: Check if folder has `marked offline files` before to call this method

File: app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt [hunk 3]
```diff
@@ -271,6 +272,8 @@ open class FileListFragment : MultiSelectFragment(MATOMO_CATEGORY), SwipeRefresh
         }
 
         setupBackActionHandler()
+
+        enqueueBulkDownloadWorker(requireContext(), folderId, userDrive ?: UserDrive())
```
Comment (sirambd) [inline app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt:276]: Check with `realm` if this forlder contains `marked offline files`

**Review by sirambd — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt (hunk 1 above)
Comment (tevincent) [reply app/src/main/java/com/infomaniak/drive/ui/MainActivity.kt]: Done

File: app/src/main/java/com/infomaniak/drive/utils/Utils.kt (hunk 2 above)
Comment (tevincent) [reply app/src/main/java/com/infomaniak/drive/utils/Utils.kt:376]: Done

File: app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt (hunk 3 above)
Comment (tevincent) [reply app/src/main/java/com/infomaniak/drive/ui/fileList/FileListFragment.kt:276]: Done

**Review by sirambd — APPROVED** [review]

**Pull request merged**